misk-core/src/test/kotlin/misk/queuing/TimedBlockingQueueTest.kt

```
@@ -0,0 +1,136 @@
+    val arrayQueue = ArrayBlockingQueue<Int>(queueSize)
+    timedQueue.addAll(values)
+    arrayQueue.addAll(values)
+
+    // read/remove operations on non-empty queue
+    val timedQueueIterator = timedQueue.iterator()
+    val arrayQueueIterator = arrayQueue.iterator()
+    while (timedQueueIterator.hasNext()) {
+      assertEquals(arrayQueueIterator.next(), timedQueueIterator.next())
+    }
+    assertFalse(arrayQueueIterator.hasNext())
+
+    for (value in values) {
+      assertEquals(arrayQueue.contains(value), timedQueue.contains(value))
+    }
+    assertEquals(arrayQueue.contains(0), timedQueue.contains(0))
+
+    assertEquals(arrayQueue.isEmpty(), timedQueue.isEmpty())
+    assertEquals(arrayQueue.remainingCapacity(), timedQueue.remainingCapacity())
+    assertEquals(arrayQueue.peek(), timedQueue.peek())
+    assertEquals(arrayQueue.element(), timedQueue.element())
+    assertEquals(arrayQueue.remove(), timedQueue.remove())
+    assertEquals(arrayQueue.poll(), timedQueue.poll())
+    assertEquals(arrayQueue.poll(1, TimeUnit.SECONDS), timedQueue.poll(1, TimeUnit.SECONDS))
+    assertEquals(arrayQueue.remove(4), timedQueue.remove(4))
+    assertEquals(arrayQueue.take(),timedQueue.take())
+    assertEquals(arrayQueue.removeAll(values), timedQueue.removeAll(values))
+    // read/remove operations on empty queue
+    assertEquals(arrayQueue.isEmpty(), timedQueue.isEmpty())
+    assertEquals(arrayQueue.remainingCapacity(), timedQueue.remainingCapacity())
+    assertEquals(arrayQueue.peek(), timedQueue.peek())
+    shouldThrowException({ timedQueue.element() }, "Exception expected")
+    shouldThrowException({ timedQueue.remove() }, "Exception expected")
+    assertEquals(arrayQueue.poll(), timedQueue.poll())
+    assertEquals(arrayQueue.poll(1, TimeUnit.SECONDS), timedQueue.poll(1, TimeUnit.SECONDS))
+  }
+
+  private fun shouldThrowException(lambda: () -> Unit, failureMsg: String) {
```

use kt `assertFailsWith`? Specifying the exception type would be nice to prevent swallowing unintended errors